src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -472,7 +477,9 @@ public static Prepared parseAndPrepare(String query, ClientState clientState, bo
             res = new Prepared(statement, "", fullyQualified, keyspace);
         else
             res = new Prepared(statement, query, fullyQualified, keyspace);
-        res.pstmntSize = measurePstmnt(res);
+
+        if (measure)
+            res.pstmntSize = measurePstmnt(res);
```

Comment:
I was worried about pstmnts with size 0 exploding the cache but `getSizeOfPreparedStatementForCache()`already guards us against that. I would add a comment though explaining sometimes we don't need to measure as it won't make it into the cache, otherwise it reads a bit strange imo,